packages/vscode-extension/src/project/project.ts

```diff
@@ -372,6 +376,52 @@ export class Project implements Disposable, MetroDelegate, ProjectInterface {
     this.deviceSession?.onActiveFileChange(filename, followEnabled);
   }
 
+  public async getCommandsCurrentKeyBinding(commandName: string) {
+    const packageJsonPath = path.join(extensionContext.extensionPath, "package.json");
+    const extensionPackageJson = require(packageJsonPath);
+    let keybindingsJsonPath;
+    let keybindingsJson;
+    try {
+      keybindingsJsonPath = path.join(
+        homedir(),
+        "Library/Application Support/Code/User/keybindings.json"
+      );
+      // can not use require because the file may contain comments
+      keybindingsJson = JSON5.parse(fs.readFileSync(keybindingsJsonPath).toString());
+    } catch (e) {
+      Logger.error("error while parsing keybindings.json", e);
+      return undefined;
+    }
+
+    const isRNIDECommand =
+      extensionPackageJson.contributes.commands &&
+      extensionPackageJson.contributes.commands.find((command: any) => {
+        return command.command === commandName;
+      });
+    if (!isRNIDECommand) {
+      Logger.warn("Trying to access a keybinding for a command that is not part of an extension.");
```

Comment:
what's the reason we're adding this restriction? I saw some other PR that was overriding the behavior of the hotkey for closing the tab, so it may sometimes be useful to read hotkeys for other commands too, no?

Reply:
I am not entirely  sure how to write a universal method, since the commands default  keybinding might be defined in multiple places (depending on if it is native for vscode or contributed by another extension), I feel like It is a significant enough challange to do it, when it is needed, a specially that I can't think of a use case for informing our users about keybindings that are not contributed by us. Right now the only use case I see is to increase discoverability of our on functionality.